OpenRA.Mods.Common/Traits/Render/RenderUpgradeRangeCircle.cs

```diff
@@ -0,0 +1,94 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Draw a circle indicating my upgrade of nearby range.")]
+	class RenderUpgradeRangeCircleInfo : ITraitInfo, IPlaceBuildingDecoration, Requires<UpgradeActorsNearInfo>
+	{
+		public readonly string RangeCircleType = null;
+
+		[Desc("Color to use")]
+		public readonly Color Color = Color.BlueViolet;
+
+		IEnumerable<UpgradeActorsNearInfo> infos = null;
+
+		public IEnumerable<UpgradeActorsNearInfo> GetInfos(ActorInfo ai)
+		{
+			if (infos == null)
+				infos = ai.Traits.WithInterface<UpgradeActorsNearInfo>()
```

Comment:
Can this check for `RenderUpgradeRangeCircleInfo` instead? I'd really like to see `UpgradeActorsNearInfo` not having a `RangeCircleType`.